src/shogun/base/AnyParameter.h

```diff
@@ -21,49 +27,102 @@ namespace shogun
 		GRADIENT_NOT_AVAILABLE = 0,
 		GRADIENT_AVAILABLE = 1
 	};
-
+	/** @brief Class AnyParameterProperties keeps track of parameter properties.
+	 *
+	 * Currently there are three types of parameters:
+	 * -# HYPERPARAMETER: the parameter determines how the training is
+	 * performed, e.g. regularisation
+	 * -# GRADIENT_PARAM: the parameter is used in gradient updates
+	 * -# MODEL_PARAM: model parameter that is trained, e.g. weight and bias
+	 * vectors and bias
+	 */
 	class AnyParameterProperties
 	{
 	public:
-		AnyParameterProperties()
-		    : m_description(), m_model_selection(MS_NOT_AVAILABLE),
-		      m_gradient(GRADIENT_NOT_AVAILABLE)
+		static const int32_t HYPERPARAMETER = 0x00000001;
+		static const int32_t GRADIENT_PARAM = 0x00000010;
```

Comment:
You can also use the << operator to shift the 1 a desired number of digits
See the stack overflow post

Reply:
I had a look at it, I do that, I just thought that the hexadecimal representation is more readable, no?

Reply:
at the end of the day, it is style. You can either define the mask as you did with hexadecimal as you did there. Or, instead, you can generate those masks on the fly using a shift operator, and then the `HYPERPARAMETER` constant is the number of bits that you shift, i.e.
```
my_mask &&
(1 << HYPERPARAMETER) &&
(1 << GRADIENT_AVAILABLE) &&
```
 The latter has the advantage that somebody reading the code never is exposed to hexadecimal numbers, nor powers or two (which can be confusing). On the other hand, masking operations (as what I wrote above) becomes a bit more notation heavy. Idk what is best to be honest. I think the shifting is more standard and therefore I think we should follow that.

Ah one thing I think is good about the shifting approach is that you can change the size of the mask easily. You are using a 32bit int above, but you only write down the last 8 bits, which is weird. Imagine you now changed the mask to 64 bit. Then the hexadecimals would be quite long, whereas the shifts wouldnt be.

@lisitsyn @vigsterkr ?

Reply:
OK! I think you're right, I have changed the class members to use bit shifts, it is a bit cleaner.